From sshtylyov@ru.mvista.com Thu Jun 21 22:24:22 2007
Return-Path: <sshtylyov@ru.mvista.com>
X-Spam-Checker-Version: SpamAssassin 3.1.7-deb (2006-10-05) on debian
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=AWL autolearn=unavailable 
	version=3.1.7-deb
Received: from imap.sh.mvista.com (unknown [63.81.120.155]) by mail.tglx.de
	(Postfix) with ESMTP id 2149065C065 for <tglx@linutronix.de>; Thu, 21 Jun
	2007 22:24:22 +0200 (CEST)
Received: from wasted.dev.rtsoft.ru (unknown [10.150.0.9]) by
	imap.sh.mvista.com (Postfix) with ESMTP id D27113EC9; Thu, 21 Jun 2007
	13:24:15 -0700 (PDT)
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Organization: MontaVista Software Inc.
To: tglx@linutronix.de, bruce.ashfield@gmail.com, rostedt@goodmis.org
Subject: [PATCH] (2.6.20-rt3) PowerPC: convert spinlocks into raw
Date: Thu, 21 Jun 2007 23:25:58 +0300
User-Agent: KMail/1.5
MIME-Version: 1.0
Content-Disposition: inline
Content-Type: text/plain; charset="iso-8859-1"
Message-Id: <200706220025.58799.sshtylyov@ru.mvista.com>
X-Evolution-Source: imap://tglx%40linutronix.de@localhost:8993/
Content-Transfer-Encoding: 8bit

Convert the spinlocks in the PowerPC interrupt related code into the raw ones,
also convert the PURR and PMC related spinlocks...

Signed-off-by: Mark A. Greer <mgreer@mvista.com>
Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>

---
Resending in hopes it still can apply -- if it doesn't, bug me again... :-)

---
 arch/powerpc/kernel/pmc.c   |    2 +-
 arch/powerpc/sysdev/i8259.c |    2 +-
 arch/powerpc/sysdev/ipic.c  |    2 +-
 arch/powerpc/sysdev/mpic.c  |    2 +-
 include/asm-powerpc/mpic.h  |    2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

Index: linux-2.6.23.1-rt5/arch/powerpc/kernel/pmc.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/powerpc/kernel/pmc.c
+++ linux-2.6.23.1-rt5/arch/powerpc/kernel/pmc.c
@@ -37,7 +37,7 @@ static void dummy_perf(struct pt_regs *r
 }
 
 
-static DEFINE_SPINLOCK(pmc_owner_lock);
+static DEFINE_RAW_SPINLOCK(pmc_owner_lock);
 static void *pmc_owner_caller; /* mostly for debugging */
 perf_irq_t perf_irq = dummy_perf;
 
Index: linux-2.6.23.1-rt5/arch/powerpc/sysdev/i8259.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/powerpc/sysdev/i8259.c
+++ linux-2.6.23.1-rt5/arch/powerpc/sysdev/i8259.c
@@ -23,7 +23,7 @@ static unsigned char cached_8259[2] = { 
 #define cached_A1 (cached_8259[0])
 #define cached_21 (cached_8259[1])
 
-static DEFINE_SPINLOCK(i8259_lock);
+static DEFINE_RAW_SPINLOCK(i8259_lock);
 
 static struct device_node *i8259_node;
 static struct irq_host *i8259_host;
Index: linux-2.6.23.1-rt5/arch/powerpc/sysdev/ipic.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/powerpc/sysdev/ipic.c
+++ linux-2.6.23.1-rt5/arch/powerpc/sysdev/ipic.c
@@ -30,7 +30,7 @@
 #include "ipic.h"
 
 static struct ipic * primary_ipic;
-static DEFINE_SPINLOCK(ipic_lock);
+static DEFINE_RAW_SPINLOCK(ipic_lock);
 
 static struct ipic_info ipic_info[] = {
 	[9] = {
Index: linux-2.6.23.1-rt5/arch/powerpc/sysdev/mpic.c
===================================================================
--- linux-2.6.23.1-rt5.orig/arch/powerpc/sysdev/mpic.c
+++ linux-2.6.23.1-rt5/arch/powerpc/sysdev/mpic.c
@@ -46,7 +46,7 @@
 
 static struct mpic *mpics;
 static struct mpic *mpic_primary;
-static DEFINE_SPINLOCK(mpic_lock);
+static DEFINE_RAW_SPINLOCK(mpic_lock);
 
 #ifdef CONFIG_PPC32	/* XXX for now */
 #ifdef CONFIG_IRQ_ALL_CPUS
Index: linux-2.6.23.1-rt5/include/asm-powerpc/mpic.h
===================================================================
--- linux-2.6.23.1-rt5.orig/include/asm-powerpc/mpic.h
+++ linux-2.6.23.1-rt5/include/asm-powerpc/mpic.h
@@ -280,7 +280,7 @@ struct mpic
 #ifdef CONFIG_MPIC_U3_HT_IRQS
 	/* The fixup table */
 	struct mpic_irq_fixup	*fixups;
-	spinlock_t		fixup_lock;
+	raw_spinlock_t		fixup_lock;
 #endif
 
 	/* Register access method */
